Packages/Netherlands3D/WebServiceReader/Runtime/Scripts/CreateWMSLayer.cs

```diff
@@ -50,135 +44,77 @@ public class WMSLOD
-            }
-            if (ShowWMSOnBuildings)
-            {
-                ShowWMSOnBuildings.AddListenerStarted(showWMSOnBuildings);
-            }
-            if (ShowWMSOnTerrain)
-            {
-                ShowWMSOnTerrain.AddListenerStarted(showWMSOnTerrain);
-            }
-            if (ShowWMSOnBuildingsAndTerrain)
-            {
-                ShowWMSOnBuildingsAndTerrain.AddListenerStarted(showWMSOnBuildingsAndTerrain);
-            }
-            if (UnloadWMSService)
-            {
-                UnloadWMSService.AddListenerStarted(UnloadLayer);
-            }
-            if (AleenOpMaaiveld_Bool)
-            {
-                AleenOpMaaiveld_Bool.AddListenerStarted(ShowOnlyOnTerrain);
-            }
-            if (ShowLayer_Bool)
-            {
-                ShowLayer_Bool.AddListenerStarted(ShowLayer);
-            }
+            tileHandler = FindObjectOfType(typeof(TileHandler)) as TileHandler;
+            if (!tileHandler)
+                Debug.LogWarning("No TileHandler found. This script depends on a TileHandler.", this.gameObject);
         }
+
         /// <summary>
         /// turn the layer on or off
         /// </summary>
         /// <param name="OnOff"></param>
         public void ShowLayer(bool OnOff)
         {
             DisplayState = OnOff;
+            Debug.Log("Show layer " + OnOff);
             if (layer)
             {
                 layer.isEnabled = OnOff;
                 layer.gameObject.SetActive(OnOff);
             }
         }
 
-        private void UnloadLayer()
+        /// <summary>
+        /// Clears layer from tilehandler
+        /// </summary>
+        public void UnloadLayer()
         {
+            Debug.Log("Removing WMS layer");
             tileHandler.RemoveLayer(layer);
             Destroy(layer.gameObject);
             layer = null;
         }
 
-        private void ShowOnlyOnTerrain(bool toggleValue)
-        {
-            
-            OnlyOnTerrain_Memory = toggleValue;
-            if (toggleValue)
-            {
-                showWMSOnTerrain();
-            }
-            else
-            {
-                showWMSOnBuildingsAndTerrain();
-            }
-        }
-
-        private void showWMSOnBuildings()
+        /// <summary>
+        /// Create a new layer using a WMS base url.
+        /// The following placeholders can be used:
+        /// {Width} and {Height} to determine requested image size.
+        /// {Xmin},{Ymin},{Xmax} and {Ymax} to set the boundingbox bottom left (xmin) and top right (ymax) coordinates.
+        /// </summary>
+        /// <param name="baseURL">The WMS base url. For example 'https://service.pdok.nl/hwh/luchtfotorgb/wms/v1_0?service=WMS&request=GETMAP&version=1.1.1&LAYERS=Actueel_orthoHR&Styles=Default&WIDTH={Width}&HEIGHT={Height}&format=image/jpeg&srs=EPSG:28992&bbox={Xmin},{Ymin},{Xmax},{Ymax}&transparent=true'</param>
+        public void CreateLayer(string baseURL)
```

Comment:
Het zou nog mooier zijn als, in plaats van een string, dit een value object zou zijn; dan hoef je niet in het DocBlock te beschrijven welke onderdelen er zijn maar, bijv, een toString gebruiken en fields hebben voor elke property. Maar ik weet niet hoeveel werk dat gaat zijn en of dat niet beter los kan?